aldryn_pypi_stats/models.py

```diff
@@ -38,6 +42,39 @@ def get_json_url(self):
     def __str__(self):
         return self.label
 
+    def get_cache_key(self):
+        """
+        Gets the cache key for this specific package configuration.
+        """
+        return get_cache_key(
+            self.__class__.__name__, settings=(self.pk, ))
+
+    def get_data(self, force_refresh=False):
+        """
+        Fetches the data (from PyPI) for this particular package configuration.
+
+        Manages a cache of the data.
+        """
+        key = self.get_cache_key()
+        if force_refresh:
+            data = None
+        else:
+            data = cache.get(key, None)
+        if not data:
+            if force_refresh:
+                logger.info('Force refresh')
+            else:
+                logger.info('Natural refresh')
+            url = self.get_json_url()
+            r = requests.get(url)
+            if r.status_code == 200:
+                data = r.json()
+            else:
+                data = None
+            duration = CACHE_DURATION * 2 if force_refresh else CACHE_DURATION
```

Comment:
a bit misleading because unless I look at the code, I would have assumed that if I say ``CACHE_DURATION`` equals x then it should expire in x. No need to take action now.